extensions/sql-database-projects/src/tools/netcoreTool.ts

```diff
@@ -96,13 +125,64 @@ export class NetCoreTool {
 		return undefined;
 	}
 
+	/**
+	 * This function checks if the installed dotnet version is 3.1 or greater.
+	 * Versions lower than 3.1 aren't supported for building projects.
+	 * Returns: True if installed dotnet version is supported, false otherwise.
+	 * 			Undefined if dotnet isn't installed.
+	 */
+	private async isNetCoreVersionSupported(): Promise<boolean | undefined> {
+		try {
+			const spawn = child_process.spawn;
+			let child: child_process.ChildProcessWithoutNullStreams;
+			let isSupported: boolean | undefined = undefined;
+
+			child = spawn('dotnet --version', [], {
+				shell: true
+			});
+
+			child.stdout.on('data', (data) => {
+				this.netCoreSdkInstalledVersion = String(data);
+			});
+
+			isSupported = await this.onExit(child);
+			if (!isSupported) {
+				this.netCoreInstallState = netCoreInstallState.netCoreVersionNotSupported;
+			}
+			return isSupported;
+		} catch (err) {
+			console.log(err);
+			return undefined;
+		}
+	}
+
+	private onExit(childProcess: child_process.ChildProcess): Promise<boolean | undefined> {
+		return new Promise((resolve, reject) => {
+			childProcess.on('exit', () => {
+				let isSupported: boolean = false;
+				if (semver.gte(this.netCoreSdkInstalledVersion!, minSupportedNetCoreVersion)) {		// Net core version greater than or equal to 3.1 are supported for Build
+					isSupported = true;
+				}
+				resolve(isSupported);
+			});
+			childProcess.on('error', (err) => {
+				console.log(err);
+				reject(undefined);
```

Comment:
You should generally be rejecting with an error, it'd be unexpected to have a rejection throw undefined